microprofile.jdt/org.eclipse.lsp4mp.jdt.core/src/main/java/org/eclipse/lsp4mp/jdt/core/PropertiesManagerForJava.java

```diff
@@ -794,4 +802,20 @@ private static int offsetOfFirstNonAnnotationModifier(BodyDeclaration node) {
 		}
 	}
 
+	/**
+	 * Returns true if the given node was generated by Project Lombok and false
+	 * otherwise.
+	 * 
+	 * @param node the node to check if it's generated
+	 * @return true if the given node was generated by Project Lombok and false
+	 *         otherwise
+	 */
+	private static boolean isGenerated(ASTNode node) {
+		try {
+			return ((Boolean) node.getClass().getField("$isGenerated").get(node)).booleanValue();
+		} catch (IllegalArgumentException | IllegalAccessException | NoSuchFieldException | SecurityException e) {

```

Comment:
Do we have to catch `IllegalArgumentException` and `SecurityException` here?

Reply:
Yes, it looks like it's required in order to access fields reflexively.